app/controllers/referee_interface/reference_controller.rb

```diff
@@ -28,16 +30,16 @@ def submit_feedback
       end
     end
 
-    def confirmation; end
-
-    def confirm_consent
-      consent_to_be_contacted = params.dig(:application_reference, :consent_to_be_contacted)
+    def submit_questionnaire
+      questionnaire_hash = ReturnQuestionnaireResultsHash.call(params: params['application_reference'])
```

Comment:
Had a thought about this and realised I think we should use our form object pattern here instead of a service. Not sure if you've come across it yet?

If not, have a look at `app/models/candidate_interface/training_with_a_disability_form.rb` and how it's used in `app/controllers/candidate_interface/training_with_a_disability_controller.rb` as an example. However, look at others as well as they all vary a bit to fit the needs of the form.

What do you think?

Reply:
Just looked. It's definitely a well-established pattern!  Will and I are going to have a refactor now.